### ISSUE TITLE
weight decay (regularization) does not work with add_loss() function

### ISSUE THREAD

**Issue by Ao-Lee** [tensorflow/tensorflow#42880]
**System information**
- OS Platform: Ubuntu 16.04
- TensorFlow installed from: conda
- TensorFlow version: 2.2.0
- Python version: 3.7
- CUDA/cuDNN version: 10.1
- GPU model and memory:  nvidia 1080

**Describe the current behavior**
I tried to add regularization loss to a model with model.add_loss() function. Normally, if I set the regularization term large enough, the regularization should damage training and give a poor result. But regularization does not affect training at all, the following code shows a toy example
```
epochs = 18
reg = 10 # set a very large regularization and see if it works
(x_train, y_train),(x_test, y_test) = tf.keras.datasets.mnist.load_data()
x_train = np.expand_dims(x_train, axis=-1) / 255.0
x_test = np.expand_dims(x_test, axis=-1) / 255.0
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])
for w in model.trainable_weights:
    if 'bias' in w.name:
        continue
    model.add_loss(lambda: tf.nn.l2_loss(w) * reg )
model.compile(optimizer='adam',loss='sparse_categorical_crossentropy',metrics=['accuracy'])
model.fit(x_train, y_train, epochs=epochs)
model.evaluate(x_test, y_test)
```

**Describe the expected behavior**
I replace the model with the following code, and it works properly:
```
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
  tf.keras.layers.Dense(128, activation='relu',  kernel_regularizer=l2(reg)),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax', kernel_regularizer=l2(reg))
])
```

**Comment by amahendrakar** [comment]
> I tried to add regularization loss to a model with model.add_loss() function. Normally, if I set the regularization term large enough, the regularization should damage training and give a poor result.

@Ao-Lee,
On adding regularization loss to the model, I did observe a decrease in the accuracy. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/7720418849b266771bde82168f149080/42880.ipynb). 

Could you please confirm if this is the behavior you are expecting. Thanks!

**Comment by Ao-Lee** [comment]
@amahendrakar amahendrakar ,

with the following code below, setting the regularization term to 10, I observe the training accuracy over one epoch is around 0.91,  which is the same result as the was no regularization.
```
reg = 10
for w in model.trainable_weights:
    if 'bias' in w.name:
        continue
    model.add_loss(lambda: tf.nn.l2_loss(w) * reg)
```

However, if the regularization is set with kernel_regularizer parameter of keras layers, I observed that the training accuracy over one epoch is around 0.11, which is what I expected. 
```
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
  tf.keras.layers.Dense(128, activation='relu',  kernel_regularizer=l2(10.0)),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax', kernel_regularizer=l2(10.0))
])
```

**Comment by Ao-Lee** [comment]
If regularization loss can not be added via model.add_loss(), it would be very hard to add regularization to pre-trained models, such as tensorflow.keras.applications.ResNet50. One temporary solution is here: https://sthalles.github.io/keras-regularizer/
I tested the methods and it correctly works. But this is not cost efficient and I hope there would be a convinent method to add regularization losses to pre-defined models

**Comment by rmothukuru** [comment]
@Ao-Lee,
This issue is duplicate of [#38891](https://github.com/tensorflow/tensorflow/issues/38891). Can you please confirm if we can close this issue as it is being tracked [there](https://github.com/tensorflow/tensorflow/issues/38891). Thanks!

**Comment by rmothukuru** [comment]
@Ao-Lee,
Can you please respond to the [above comment](https://github.com/tensorflow/tensorflow/issues/42880#issuecomment-730431019). Thanks!
